neo/rawio/maxwellrawio.py

```diff
@@ -75,21 +78,33 @@ def _parse_header(self):
         elif int(version) > 20160704:
             # multi stream stream (one well is one stream)
             self._old_format = False
-            stream_ids = list(h5["wells"].keys())
-            for stream_id in stream_ids:
-                rec_names = list(h5["wells"][stream_id].keys())
-                if len(rec_names) > 1:
-                    if self.rec_name is None:
-                        raise ValueError(
-                            "Detected multiple recordings. Please select a "
-                            "single recording using the `rec_name` parameter. "
-                            f"Possible rec_name {rec_names}"
-                        )
+            well_ids = list(h5file["wells"].keys())
+            unique_rec_names = []
+            for well_name in well_ids:
+                rec_names = list(h5file["wells"][well_name].keys())
+                for rec_name in rec_names:
+                    unique_rec_names.append(rec_name)
+            # check consistency of rec_names
+            unique_rec_names = np.unique(unique_rec_names)
```

Comment:
```suggestion
            unique_rec_names = list(set(unique_rec_names))
```
Since unique_req_names is a list set is actually way faster. Not that we need to over-optimize. But just as an fyi. :) 

```python
In [11]: %timeit set(test_list)
152 ns ± 0.118 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [12]: import numpy as np

In [13]: %timeit np.unique(test_list)
4.65 µs ± 15.2 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [17]: %timeit list( set(test_list)) # just because later in the code you need to convert here is the comparison adding in the list conversion
249 ns ± 2.06 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

Reply:
Won't make a big difference here, but good to know @zm711! I wasn't aware :)